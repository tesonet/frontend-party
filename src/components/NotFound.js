import React from "react";
import { Typography, makeStyles } from "@material-ui/core";

const useStyles = makeStyles(theme => ({
  "@global": {
    body: {
      WebkitOverflowScrolling: "touch",
      overflow: "hidden",
      backgroundImage: "url('../resources/404.jpg')",
      backgroundRepeat: "no-repeat",
      zoom: 1.5,
      backgroundPosition: "center center",
      backgroundSize: "cover",
      position: "fixed",
      left: 0,
      right: 0,
      height: "100%"
    },
    html: {
      height: "100%"
    },
    "#componentWithId": {
      height: "100%"
    }
  },
  paper: {
    display: "flex",
    height: "100%",
    flexDirection: "column",
    alignItems: "center"
  },
  notFound: {
    marginTop: theme.spacing(12),
    color: "#fff",
    fontSize: 96,
    letterSpacing: 10,
    fontWeight: 500
  },
  notFoundSmall: {
    color: "#fff",
    fontSize: 12,
    letterSpacing: 1,
    textAlign: "center"
  }
}));

export default function NotFound() {
  const classes = useStyles();
  return (
    <div className={classes.paper}>
      <Typography className={classes.notFound}>404</Typography>
      <Typography className={classes.notFoundSmall}>
       The page you requested could not be found ü§∑‚Äç‚ôÇÔ∏è
      </Typography>
    </div>
  );
}
