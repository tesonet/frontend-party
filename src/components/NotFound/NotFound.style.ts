import styled from 'styled-components'

export const NotFoundWrapper = styled.div`
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`
