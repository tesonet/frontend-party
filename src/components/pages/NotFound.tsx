import React from "react";
import styled from "@emotion/styled";
import Colors from "../../constants/colors";
import { RouteComponentProps } from "@reach/router";

const StyledBackground = styled.div`
  background-color: ${Colors.white};
  align-content: center;
  justify-content: center;
  align-items: center;
  background-size: cover;
  width: 100%;
  height: 100%;
  padding: 28px;
`;

const StyledGif = styled.img`
  height: 500px;
  width: 500px;
`;

const NotFound: React.FC<RouteComponentProps> = () => {
  return (
    <StyledBackground>
      <StyledGif src="https://i.imgur.com/gBxP6oJ.gif" />
    </StyledBackground>
  );
};

export default NotFound;
