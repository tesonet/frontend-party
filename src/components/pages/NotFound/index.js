import React from 'react'
import styled from 'styled-components'
import PageTemplate from '../../templates/Page'
import DashboardBar from '../../common/DashboardBar'
import Footer from '../../common/Footer'
import NotFoundImage from '../../../assets/404.gif'

const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

const ErrorCode = styled.h1`
  color: red;
  text-shadow: black 1px 2px;
  font-size: 70px;
  text-align: center;
`

const NotFound = () => {
  return (
    <PageTemplate header={<DashboardBar />} footer={<Footer />}>
      <Content>
        <ErrorCode>404</ErrorCode>
        <img
          src={NotFoundImage}
          alt="Now you're lost, Lost in the heat of it all"
        />
      </Content>
    </PageTemplate>
  )
}

export default NotFound
