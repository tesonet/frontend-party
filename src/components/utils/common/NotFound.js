import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../../../assets/img/testio_color.png';

const NotFound = () => (
    <section className="not-found">
        <div className="not-found__container">
            <img src={logo} alt="Logo" className="not-found__image" />
            <h2 className="not-found__title">Ooops.</h2>
            <p className="not-found__description">
                The page you are loooking for does not exist
            </p>
            <Link to="/" className="not-found__link">
                Back to application
            </Link>
        </div>
    </section>
);

export default NotFound;
