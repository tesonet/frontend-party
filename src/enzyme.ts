import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
