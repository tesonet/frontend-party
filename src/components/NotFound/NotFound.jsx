import React from "react";
import "./NotFound.scss";

const NotFound = () => <h2>Sorry, nothing here</h2>;

export default NotFound;
