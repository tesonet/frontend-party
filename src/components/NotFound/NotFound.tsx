import Button from 'components/core/Button/Button'
import React from 'react'
import { useHistory, useLocation } from 'react-router'
import { NotFoundWrapper } from './NotFound.style'

const NotFound: React.FC = () => {
  const location = useLocation()
  const history = useHistory()
  const content = `The page ${location.pathname} was not found in this party`

  const returnToHome = () => {
    history.push('/')
  }

  return (
    <NotFoundWrapper>
      <div>{content}</div>
      <Button big title="Go to main page" onClick={returnToHome} />
    </NotFoundWrapper>
  )
}

export default NotFound
