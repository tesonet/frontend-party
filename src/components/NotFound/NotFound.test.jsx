import * as React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import NotFound from "./NotFound";

configure({ adapter: new Adapter() });

describe("NotFound component", () => {
    it("should render component correctly", () => {
        const wrapper = shallow(<NotFound />);
        expect(wrapper).toMatchSnapshot();
    });
});
