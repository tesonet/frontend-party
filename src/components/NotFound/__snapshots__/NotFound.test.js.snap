// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render component correctly 1`] = `ShallowWrapper {}`;
